parrot/recorder_test.go

```diff
@@ -0,0 +1,137 @@
+package parrot
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestResponseWriterRecorder(t *testing.T) {
```

Comment:
I think it's worth adding a test case for parallel usage, in current version in `CTFv2` we have a mutex in our recorder

Reply:
Agreed, adding